src/nucname.cpp

```diff
@@ -586,12 +598,17 @@ int pyne::nucname::id(std::string nuc) {
 
     // Figure out if we are meta-stable or not
     std::string end_char = pyne::last_char(nucstr);
-    if (end_char == "M")
-      newnuc = (10000 * anum) + 1;
-    else if (pyne::contains_substring(pyne::digits, end_char))
-      newnuc = (10000 * anum);
-    else
+    size_t pos = upper_metastable.find(end_char);
+
+    if (pos != std::string::npos) {
+        newnuc = (10000 * anum) + static_cast<int>(pos) + 1;
+    }
+    else if (pyne::contains_substring(pyne::digits, end_char)) {
+        newnuc = (10000 * anum);
+    }
+  else {
       throw NotANuclide(nucstr, newnuc);
+  }
```

Comment:
Maybe this can be a new function:
```
int nucname::isomer_id(std::string end_char) {
    
    int id = -1;
    size_t pos = pyne::nucname::metastable_states.find(std::tolower(end_char));

    if (pos != std::string::npos) {
        id = static_cast<int>(pos) + 1;
    }
    else if (pyne::contains_substring(pyne::digits, end_char)) {
        id = 0;
    }

    return id;
}
```

and then:

```
isomer_id = nucname::isomer_id(end_char);
if isomer_id > -1 {
     newnuc = (10000 * anum) + isomer_id(end_char, newnuc);
}
else {
    throw NotANuclide(nucstr, newnuc);
}
```